File: unit-tests/java-junit/src/main/java/com/qatraining/ShoppingCart.java
package com.qatraining;

import java.util.ArrayList;
import java.util.List;

public class ShoppingCart {

    private static class Item {
        String name;
        double price;
        int quantity;

        Item(String name, double price, int quantity) {
            this.name = name;
            this.price = price;
            this.quantity = quantity;
        }
    }

    private final List<Item> items = new ArrayList<>();

    public void addItem(String name, double price, int quantity) {
        if (price < 0) {
            throw new IllegalArgumentException("Price cannot be negative");
        }
        if (quantity <= 0) {
            throw new IllegalArgumentException("Quantity must be positive");
        }
        items.add(new Item(name, price, quantity));
    }

    public void removeItem(String name) {
        items.removeIf(item -> item.name.equals(name));
    }

    public double getTotal() {
        return items.stream()
                .mapToDouble(item -> item.price * item.quantity)
                .sum();
    }

    public int itemCount() {
        return items.stream().mapToInt(item -> item.quantity).sum();
    }

    public boolean isEmpty() {
        return items.isEmpty();
    }
}

File: unit-tests/python-pytest/shopping_cart.py
class ShoppingCart:
    def __init__(self):
        self.items = []

    def add_item(self, name, price, quantity=1):
        if price < 0:
            raise ValueError("Price cannot be negative")
        if quantity <= 0:
            raise ValueError("Quantity must be positive")
        self.items.append({"name": name, "price": price, "quantity": quantity})

    def remove_item(self, name):
        self.items = [item for item in self.items if item["name"] != name]

    def get_total(self):
        return sum(item["price"] * item["quantity"] for item in self.items)

    def item_count(self):
        return sum(item["quantity"] for item in self.items)

    def is_empty(self):
        return len(self.items) == 0